community/modules/file-system/gke-persistent-volume/variables.tf

```diff
@@ -14,6 +14,38 @@
  * limitations under the License.
 */
 
+variable "cluster_id" {
+  description = "An identifier for the GKE cluster in the format `projects/{{project}}/locations/{{location}}/clusters/{{cluster}}`"
+  type        = string
+}
+
+variable "network_storage" {
+  description = "Network attached storage mount to be configured."
+  type = object({
+    server_ip             = string,
+    remote_mount          = string,
+    local_mount           = string,
+    fs_type               = string,
+    mount_options         = string,
+    client_install_runner = map(string)
+    mount_runner          = map(string)
+  })
+}
+
+variable "filestore_id" {
```

Comment:
As this module develops, is your intent to require that the user supply exactly 1 Filestore ID _or_ 1 bucket _or_ 1 filesystem-to-be-named-later?

Reply:
Correct. There is a 1:1 between filesystem and gke-persistent-volume. A future version would have filestore_id, and some other variable for bucket, they would both default to null and a validation would be done to ensure one of them was set. Probably it is technically incorrect for this version to allow null as in input in validation.

Reply:
The reason I think it merits consideration now is that I did some quick analyses of how you might ensure that only one of the inputs values was set and could not find a good model for using `precondition` to prevent it.  I did have the thought that you might eliminate the problem by simply allowing multiple PVs/PVCs per instance of this module:

1. add an `s` to the names
2. allow the module to combine buckets and Filestore and X/Y/Z into a set of PVs/PVCs

Might avoid a lot of user and developer frustration. I also came across this today:

https://developer.hashicorp.com/terraform/language/functions/try

Look at "We can also use try to deal..." to see how you might treat input Filestore IDs as a string or a list so that you can have N>1 Filestore/Buckets per module, too.

Reply:
I think the thing to do right now is:

- disallow `var.filestore_id == null`; that appears to generate genuinely broken inputs
- add the `s`

Consider the comments above to evolve the design.

Reply:
I think I have satisfied your short term requests. I will also look at how to control the "choose one" issue as it is clearly not super straight forward.